### ISSUE TITLE
Build creates "~/go" directory

### ISSUE THREAD

**Issue by jpfkd** [Jguer/yay#1047]
#### Affected Version
`yay v9.3.2.r7.g769aee7 - libalpm v11.0.3`

#### Issue
When I build yay, a directory named "go" is created in my home directory.  Many files in this directory don't have write access of the owner.
Running makepkg on the AUR package "yay 9.3.2-1" causes this issue as well.

#### Steps to reproduce
```
git clone https://github.com/Jguer/yay
cd yay
make build
```
or
```
git clone https://aur.archlinux.org/yay.git
cd yay
makepkg
```

#### Output
The output of `ls -lR go` (junpei is my user name.): https://paste.debian.net/1104601/

**Comment by Jguer** [comment]
```
git clone https://github.com/Jguer/yay
cd yay
make build
```
Tried it, didn't work

```
git clone https://aur.archlinux.org/yay.git
cd yay
makepkg
```
Got it to work.

It's a duplicate of https://github.com/Jguer/yay/issues/1032

But since you can reproduce it on the git clone can you reply there if you have any go related environment variables?

**Comment by Jguer** [comment]
Not a duplicate but closely linked #1032.

https://golang.org/cmd/go/#hdr-GOPATH_and_Modules

This is what's happening, I just need to figure out if this is suppose to happen when you have vendored dependencies or not. 

Off-topic: If you don't use the go programming language, `yay-bin` could be a better choice

**Comment by Jguer** [comment]
Fixed in fdcfeba9024b035d2b094def0053ed4af151e338.

While searching for source files, find would ignore all the go files in the vendor directory but not the vendor directory itself.

```
$(BIN): $(SOURCES)
	$(GO) build $(GOFLAGS) -ldflags '-s -w $(LDFLAGS)' $(EXTRA_GOFLAGS) -o $@
```

would then turn to 
yay: ./a.go ./b.go ./vendor ...

which would then call 
```
vendor:
	$(GO) mod tidy && $(GO) mod vendor
```
that does indeed download the deps and creates the `$GOPATH/pkg`, mistery solved
